sdk/core/azure-core/src/main/java/com/azure/core/util/RequestContent.java

```diff
@@ -0,0 +1,248 @@
+     */
+    static RequestContent create(byte[] bytes, int offset, int length) {
+        Objects.requireNonNull(bytes, "'bytes' cannot be null.");
+        return new ArrayContent(bytes, offset, length);
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses {@link String} as its data.
+     * <p>
+     * The passed {@link String} is converted using {@link StandardCharsets#UTF_8}, if another character set is required
+     * use {@link #create(byte[])} and pass {@link String#getBytes(Charset)} using the required character set.
+     *
+     * @param content The string that will be the {@link RequestContent} data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code content} is null.
+     */
+    static RequestContent create(String content) {
+        Objects.requireNonNull(content, "'content' cannot be null.");
+        return create(content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses {@link BinaryData} as its data.
+     *
+     * @param content The {@link BinaryData} that will be the {@link RequestContent} data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code content} is null.
+     */
+    static RequestContent create(BinaryData content) {
+        Objects.requireNonNull(content, "'content' cannot be null.");
+        return new ByteBufferContent(content.toByteBuffer());
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses {@link Path} as its data.
+     *
+     * @param file The {@link Path} that will be the {@link RequestContent} data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code file} is null.
+     * @throws UncheckedIOException If the size of the {@code file} cannot be determined.
+     */
+    static RequestContent create(Path file) {
+        Objects.requireNonNull(file, "'file' cannot be null.");
+        try {
+            return create(file, 0, Files.size(file));
+        } catch (IOException e) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new UncheckedIOException(e));
+        }
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses {@link Path} as its data.
+     *
+     * @param file The {@link Path} that will be the {@link RequestContent} data.
+     * @param offset Offset in the {@link Path} where the data will begin.
+     * @param length Length of the data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code file} is null.
+     */
+    static RequestContent create(Path file, long offset, long length) {
+        Objects.requireNonNull(file, "'file' cannot be null.");
+        return new FileContent(file, offset, length);
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses a serialized {@link Object} as its data.
+     * <p>
+     * This uses an {@link ObjectSerializer} found on the classpath.
```

Comment:
Add additional documentation that this won't have a determinable `getLength`, so if it is required then use ObjectSerializer directly and create a `byte[]` `RequestContent`.